astropy/io/ascii/ui.py

```diff
@@ -593,6 +621,118 @@ def _get_guess_kwargs_list(read_kwargs):
     return guess_kwargs_list
 
 
+def _read_in_chunks(table, **kwargs):
+    """
+    For fast_reader read the ``table`` in chunks and vstack to create
+    a single table, OR return a generator of chunk tables.
+    """
+    fast_reader = kwargs['fast_reader']
+    chunk_size = fast_reader.pop('chunk_size')
+    chunk_generator = fast_reader.pop('chunk_generator', False)
+    fast_reader['parallel'] = False  # No parallel with chunks
+
+    tbl_chunks = _read_in_chunks_generator(table, chunk_size, **kwargs)
+    if chunk_generator:
+        return tbl_chunks
+
+    tbl0 = next(tbl_chunks)
+    masked = tbl0.masked
+
+    # Numpy won't allow resizing the original so make a copy here.
+    out_cols = {col.name: col.data.copy() for col in tbl0.itercols()}
+
+    for tbl in tbl_chunks:
+        col_len = len(tbl)
+        masked |= tbl.masked
+        for name, col in tbl.columns.items():
+            if masked:
+                # For masked case, concatenate old + new columns
+                out_cols[name] = np.ma.concatenate([out_cols[name], col.data])
+            else:
+                # For unmasked do an inplace resize and assignment (saves memory?)
+                out_col = out_cols[name]
+                out_col.resize(len(out_col) + col_len, refcheck=False)
+                out_col[-col_len:] = col.data
+
+    # Make final table from numpy arrays, converting dict to list
+    out_cols = [out_cols[name] for name in tbl0.colnames]
+    out = tbl0.__class__(out_cols, names=tbl0.colnames, meta=tbl0.meta,
+                         copy=False)
+
+    return out
+
+
+def _read_in_chunks_generator(table, chunk_size, **kwargs):
+    """
+    For fast_reader read the ``table`` in chunks and return a generator
+    of tables for each chunk.
+    """
+
+    @contextlib.contextmanager
+    def passthrough_fileobj(fileobj, encoding=None):
+        """Stub for get_readable_fileobj, which does not seem to work in Py3
+        for input File-like object, see #6460"""
+        yield fileobj
+
+    # Set up to coerce `table` input into a readable file object by selecting
+    # an appropriate function.
+
+    # Convert table-as-string to a File object.  Finding a newline implies
+    # that the string is not a filename.
+    if (isinstance(table, str) and ('\n' in table or '\r' in table)):
```

Comment:
I don't know much about the fast readers. Can this be hardcoded here? In the python readers one could define a splitter that used other characters to split, but I'm assuming the fast readers don't allow that, right?

Reply:
This code is basically the same as [here in core.py](https://github.com/astropy/astropy/blob/5d18c3f8cbc0839b21faa640780b0d78c120e742/astropy/io/ascii/core.py#L290) and is just about figuring out what `table` contains (filename, file object, or table string).

Indeed you cannot define custom splitters in the fast reader, splitting is all done in C-code.